cetmix_tower_server/models/cx_tower_secret_mixin.py

```diff
@@ -0,0 +1,60 @@
+from odoo import api, fields, models
+
+
+class CxTowerSecretMixin(models.AbstractModel):
+    _name = "cx.tower.secret.mixin"
+    _description = "Mixin for managing secrets"
+
+    secret_ids = fields.Many2many(
+        comodel_name="cx.tower.key",
+        relation="cx_tower_secret_rel",
+        column1="res_id",
+        column2="secret_id",
+        string="Secrets",
+        readonly=True,
+        store=True,
+        compute="_compute_secret_ids",
+        groups="cetmix_tower_server.group_manager,cetmix_tower_server.group_root",
+    )
+
+    @api.depends("code")
+    def _compute_secret_ids(self):
+        """
+        Compute the secret IDs based on the references found in the code field.
+        """
+
+        for record in self:
+            if record.code:
+                # Use a set for faster membership testing
+                secret_ids = self._extract_secret_ids(record.code)
+                record.secret_ids = [(6, 0, secret_ids)]
+            else:
+                record.secret_ids = [(6, 0, [])]
+
+    @api.model
+    def _extract_secret_ids(self, code):
+        """
+        Extract secret IDs based on references found in the given `code`.
+
+        Args:
+            code: Text containing potential secret references.
+
+        Returns:
+            list: List of secret IDs corresponding to the references in `code`.
+        """
+        if not code:
+            return []
+
+        # Pre-fetch all secrets and map references to IDs
+        secrets = self.env["cx.tower.key"].search([])
```

Comment:
What if there are 100000 million secrets in the db? Does it mean we need to fetch all of them every time?